keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1950 @@
+container-level values are not set. As a result, the CA won't scale the number of
+nodes to accommodate this pod. Meanwhile, the scheduler will evaluate the
+pod-level resource requests but may be unable to find a suitable node to fit the
+pod. Consequently, the pod will not be scheduled. While this behavior is
+acceptable for the alpha implementation, it is anticipated that Cluster
+Autoscaler support will be addressed in the Beta phase with pod resource
+requirements surfaced in a helper library/function that autoscalers can use to
+make autoscaling decisions.
+
+#### [Scoped for Beta] Support for Windows
+
+Pod-level resource specifications are a natural extension of Kubernetes' existing
+resource management model. Although this new feature is expected to function with
+Windows containers, careful testing and consideration are required due to
+platform-specific differences. As the introduction of pod-level resources is a
+major change in itself, full support for Windows will be addressed in future
+stages, beyond the initial alpha release.
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes
+necessary to implement this enhancement.
+
+#### Unit tests
+
+This feature will touch multiple components. For alpha, unit tests coverage for following packages needs to be added:
+
+* Scheduler logic will be updated to consider pod-level requests. Hence pkg/scheduler will require additional coverage.
+* pkg/kubelet/cm will be updated to set pod-level cgroups using CPU requests and limits, and memory limits.
+* pkg/apis/core/validation/types_test.go and pkg/apis/core/validation since new fields are added in PodSpec and also new validation rules are required for the new fields.
+* pkg/kubeapiserver/admission for changes made in the admission logic for LimitRanger and ResourceQuota admission controllers.
+
+
+#### e2e tests
+
+Following scenarios need to be covered:
+
+
+
+* Cgroup settings when pod-level resources are set.
+* Validate scheduling and admission.
+* Validate the containers with no limits set are throttled on CPU when CPU usage reaches Pod level CPU limits.
+* Validate the containers with no limits set are OOMKilled when memory usage reaches Pod level memory limits.
+
+
+### Graduation Criteria
+
+
+#### Phase 1: Alpha (target 1.31)
+
+
+* Feature is disabled by default. It is an opt-in feature which can be enabled by enabling the `PodLevelResources`
+feature gate and by setting the new `resources` fields in PodSpec at Pod level.
+* Support the basic functionality for scheduler to consider pod-level resource requests to find a suitable node.
+* Support the basic functionality for kubelet to translate pod-level requests/limits to pod-level cgroup settings.
+* Unit test coverage.
+* E2E tests.
+* Documentation mentioning high level design.
+
+
+#### Phase 2:  Beta (target 1.32)
```

Comment:
so things like cpu manager, limit ranger, etc. are not part of beta graduation criteria? Or those are in alpha?

Reply:
Support for all features that are scoped out of alpha phase in the KEP design includes CPU manager. 

There's a section for Limit Ranger in alpha.